FFmpegInterop/Source/UncompressedAudioSampleProvider.cpp

```diff
@@ -134,7 +138,14 @@ MediaStreamSample^ UncompressedAudioSampleProvider::GetNextSample()
 			{
 				finalPts = pts;
 			}
-			finalDur += dur;
+			if (dur == 0 && dataWriter->UnstoredBufferLength != 0)
+			{
+				finalDur = (dataWriter->UnstoredBufferLength / avgBytesPerSec) * 10000000;
```

Comment:
I think @khouzam means this line. Maybe better do it like this?

`finalDur = ((long long)dataWriter->UnstoredBufferLength * 10000000) / avgBytesPerSec`

I am not sure if the long long cast is neccessary, but we need to be sure that we don't get overflow.

Reply:
I will change all of them this weekend. I haven't had the time to do much testing and development on this lately :(